app/frontend/src/features/auth/jail/Jail.tsx

```diff
@@ -0,0 +1,59 @@
+import { Backdrop, makeStyles } from "@material-ui/core";
+import React, { useEffect, useState } from "react";
+import { Redirect } from "react-router-dom";
+import Alert from "../../../components/Alert";
+import CircularProgress from "../../../components/CircularProgress";
+import PageTitle from "../../../components/PageTitle";
+import TextBody from "../../../components/TextBody";
+import { JailInfoRes } from "../../../pb/jail_pb";
+import { getJailInfo } from "../../../service/jail";
+import { useAppDispatch, useTypedSelector } from "../../../store";
+import { updateJailStatus } from "./jailActions";
+import TOSSection from "./TOSSection";
+
+const useStyles = makeStyles((theme) => ({
+  bottomMargin: { marginBottom: theme.spacing(4) },
+}));
+
+export default function Jail() {
+  const classes = useStyles(makeStyles);
+
+  const dispatch = useAppDispatch();
+  const isJailed = useTypedSelector((state) => state.auth.jailed);
+  const authError = useTypedSelector((state) => state.auth.error);
+  const authLoading = useTypedSelector((state) => state.auth.loading);
+
+  const [loading, setLoading] = useState(false);
+  const [jailInfo, setJailInfo] = useState<null | JailInfoRes.AsObject>(null);
+
+  useEffect(() => {
+    (async () => {
+      //just in case the store is stale
+      dispatch(updateJailStatus());
+      setLoading(true);
+      setJailInfo(await getJailInfo());
+      setLoading(false);
```

Comment:
Kept loading because getJailInfo doesn't happen in the redux store.

Reply:
Maybe it should happen there?

Reply:
Yeah maybe. I'm going to merge this for now since some other pulls are waiting on this and I'll see if creating a jail reducer is a better solution when the next "jail condition" comes in, which will be locations